### PR TITLE
Add language switcher functionality

### PR DESCRIPTION
**Pull request by AlexandruBuhai**
Closes: #97

This pr focuses on the language change functionality for our project. I chose the Lingui Js library since it seems to offers the best features/ease of use. Right now the I pass the callbacks for changing the language using props. If we ever change to Redux or another state management system this can be improved.

**One thing that I still have to do in order to complete this feature is to add documentation and a how-to guide.**

Also, I had no design on where to put the language switcher, so I want to get feedback on that (I can change it to the other side of the buttons, flags instead of text, etc.). How it looks right now:
Screenshot 2020-02-01 at 17 04 44
Screenshot 2020-02-01 at 17 04 29

Edit: after this PR merges, I will create the translation file for all the text in the repo.

### CODE REVIEW

**Comment by tudoramariei** [comment]
@AlexandruBuhai use `Fixes` or `Closes` and then the `#issue_number` so that GitHub will add an actual link between the PR and issue (i.e. `Fixes #97`)

**Comment by tudoramariei** [comment]
I'll need a bit of thinking here since we're extending the scope of this task right now. The initial plan was to just internationalize the application. I don't think the language switcher a bad feature, just that it will bring some extra work on the back-end side and I'm not sure we might need it (for the initial deployment or any others).
I think I'll need to get an opinion from @catileptic and @RaduCStefanescu on this.
Also, if @danniel (who is working on back-end i18n) would be OK with the extra work.

**Comment by AlexandruBuhai** [comment]
Then I got something wrong with `the internationalization of the application task and the suggestion of react-i18next`. Cause react-i18next has the same purpose as lingui as far as I know, extract all the text, to be easily translated and changed later. The language switcher is optional and I understand it may be more work on the backend with some stuff, but it shouldn't be that much depending on what responses we send to the front end. Let me know anytime what the resolution is, and I'll make the according chages. Thanks for looking over this!

**Comment by tudoramariei** [comment]
@AlexandruBuhai after some discussions, the conclusion is that I'll need you to resolve your conflicts and ping me when you're ready
The language switcher is a go! 🙌

File: client/src/locales/en/messages.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "Adaugă o clădire": {
+    "translation": "Add a building",
+    "origin": [
+      [
+        "src/components/Header.js",
+        21
+      ]
+    ]
+  },
```
[the remark this answers is not in the document]
Comment (AlexandruBuhai) [reply client/src/locales/en/messages.json]: The keys are automatically generated from the text inside the `<Trans />` component, so for us to have English keys, we would need to write all the text (that we want to be translated) in English. This is no problem for short text like the buttons, but could become tedious for Romanian programmers to also translate the let's say Legal Part of our website, such that when they make the PR, the `messages.json` should be correct and 'final'.

[the remark this answers is not in the document]
Comment (AlexandruBuhai) [reply client/src/locales/en/messages.json]: That being said, I will make the desired change to English default language for keys, but the website will open in Romanian. But we have to not make the translation compulsatory, if we ever add a Github PR check for this.

**Pull request merged**